File: preprocess.lua
-- pre-process and save data
require 'mattorch'
require 'sys'

TOTALFACES = 5231
num_train_batches = 5000
num_test_batches =  TOTALFACES-num_train_batches-1

bsize = 50
imwidth = 150

function load_batch(id)
	collectgarbage()
  basedir = '../facemachine/CNN_DATASET/'
  local t = mattorch.load(basedir .. 'face_' .. id .. '/data' .. '.mat' )
  t = t.img:reshape(bsize, 3, imwidth, imwidth):float()/255
  t = torch.Tensor(t:size()):copy(t)
  return t
end


for i=1,num_train_batches do
 	local t = load_batch(i)
 	print(i, '/', num_train_batches)
 	torch.save('DATASET/training/batch' .. i)
end

for i=1,num_test_batches do
  local FID = i + num_train_batches
  local t = load_batch(FID)
  print(i, '/', num_test_batches)
 	torch.save('DATASET/test/batch' .. i)
end

File: preprocess.m
TOTALFACES = 0;%5230;
num_train_batches = 200;%5000;
num_test_batches =  0;%TOTALFACES-num_train_batches;

for i=1:num_train_batches
   basedir = '../facemachine/CNN_DATASET_FINET/';
%    load([basedir 'face_' num2str(i) '/data.mat']);
%    img = double(img)/255.0;
%    save(['DATASET/training/' num2str(i) '.mat'], 'img');
  allimg = dir([basedir 'face_' num2str(i) '/*.png']);
  cnt=1;
 mkdir(['DATASET/FT_training/face_' num2str(i)]); 
  for f=allimg'
     fname = f.name;
     img = imread([basedir 'face_' num2str(i) '/' fname]);
     imwrite(img,['DATASET/FT_training/face_' num2str(i) '/' num2str(cnt) '.png']);
     cnt = cnt + 1;
  end
    i
end

for i=1:num_test_batches
   basedir = '../facemachine/CNN_DATASET/';
%    load([basedir 'face_' num2str(i) '/data.mat']);
%    img = double(img)/255.0;
%    save(['DATASET/training/' num2str(i) '.mat'], 'img');
  allimg = dir([basedir 'face_' num2str(i) '/*.png']);
  cnt=1;
 mkdir(['DATASET/FT_test/face_' num2str(i)]); 
  for f=allimg'
     fname = f.name;
     img = imread([basedir 'face_' num2str(i) '/' fname]);
     imwrite(img,['DATASET/FT_test/face_' num2str(i) '/' num2str(cnt) '.png']);
     cnt = cnt + 1;
  end
    i
end